sopel/plugins/__init__.py

```diff
@@ -67,12 +67,8 @@ def find_internal_plugins():
     Internal plugins can be found under ``sopel.modules``. This list does not
     include the ``coretasks`` plugin.
     """
-    plugin_dir = imp.find_module(
-        'modules',
-        [imp.find_module('sopel')[1]]
-    )[1]
-
-    for name, _ in _list_plugin_filenames(plugin_dir):
+    modules = importlib.import_module('sopel.modules')
+    for name, _ in _list_plugin_filenames(modules.__path__[0]):
```

Comment:
`modules.__path__[0]` feels a bit voodoo-y, but I don't understand importlib enough to suggest an alternative.

Reply:
Yup, you have to read more on that in the Python's documentation: https://docs.python.org/3/reference/import.html#module-path

Only package module should have a path, and that path must be an iterable. In theory, there should be one and only one path for `sopel.modules.__path__`, hence the `[0]`.

Reply:
I meant that _plus_ the private `_list_plugin_filenames()` helper function. Lazy typing on mobile reviews, lol.

Read that page before, and understood why this works, but the use of dunder attributes feels icky somehow. 🤷‍♂️ Then again, in hindsight, what this replaces wasn't much better with its multiple nested `[1]` subscripts.

Really, it just _seems_ like something that should be doable in a prettier, more scrutable way. That said, the available functions in `importlib` and friends don't look like they'd help make it prettier. Maybe `importlib.util.find_spec()` (which gives you a [`ModuleSpec`](https://docs.python.org/3/library/importlib.html#importlib.machinery.ModuleSpec) with `submodule_search_locations` attached) would be a step in the right direction. Doing it that way at least resembles an API of some kind instead of reaching into the module for magic attrs, and skips the weird-looking use of `import_module()` when we don't actually want or need to do that.

How'd I do for giving you ideas? Good enough for skimming some docs during a video meeting? 😁